src/main/java/nl/andrewl/emaildatasetbrowser/view/ResponsiveJText.java

```diff
@@ -0,0 +1,103 @@
+package nl.andrewl.emaildatasetbrowser.view;
+
+import javax.swing.JTextArea;
+import javax.swing.JTextField;
+
+import java.awt.event.KeyEvent;
+import java.awt.event.KeyListener;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.function.Consumer;
+
+/**
+ * Implements generalized listener for JTextField and JTextArea components.
+ * Allowing you to only listen to the events you're interested in.
+ */
+public class ResponsiveJText implements KeyListener {
+    private record Listener(int key, int keyMask, Consumer<KeyEvent> event) {
+    }
+
+    public enum KeyEventType {
+        KEY_TYPED,
+        KEY_PRESSED,
+        KEY_RELEASED
+    }
+
+    private final int DEFAULT_KEY = KeyEvent.VK_ENTER;
+    private final int DEFAULT_MASK = KeyEvent.CTRL_DOWN_MASK;
+
+    private final HashMap<KeyEventType, ArrayList<Listener>> listeners = new HashMap<>();
+
+    private ResponsiveJText() {
+        for (KeyEventType key : KeyEventType.values()) {
+            listeners.put(key, new ArrayList<>());
+        }
+    }
+
+    public ResponsiveJText(JTextArea textArea) {
+        this();
+        textArea.addKeyListener(this);
+    }
+
+    public ResponsiveJText(JTextField textField) {
+        this();
+        textField.addKeyListener(this);
+    }
```

Comment:
I don't think it's a good idea to have these constructors, because their only purpose seems to be obscuring the call to `addKeyListener`. It would make for more readable code if instead of this:
```java
JTextField myField = new JTextField();
new ResponsiveJText(myField).addKeyListener((e) -> print());
```
we used this, or something like it:
```java
JTextField myField = new JTextField();
myField.addKeyListener(new ResponsiveJTextKeyListener((e) -> print()));
```
With the former, it appears to a reader (who is not 100% familiar with the implementation) that we are simply constructing a new object and immediately discarding it so it can be removed by GC, since your average reader is not expecting side-effects in a constructor usually.